packages/upgrade/src/actions/install.ts

```diff
@@ -0,0 +1,125 @@
+import type { Context, PackageInfo } from './context.js';
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { color, say } from '@astrojs/cli-kit';
+import { pluralize, celebrations, done, error, info, log, spinner, success, upgrade, banner, title, changelog, warn, bye } from '../messages.js';
+import { shell } from '../shell.js';
+import { random, sleep } from '@astrojs/cli-kit/utils';
+
+export async function install(
+	ctx: Pick<Context, 'version' | 'packages' | 'packageManager' | 'prompt' | 'dryRun' | 'exit' | 'cwd'>
+) {
+	await banner();
+	log('')
+	const { current, dependencies, devDependencies } = filterPackages(ctx);
+	const toInstall = [...dependencies, ...devDependencies];
+	for (const packageInfo of current.sort(sortPackages)) {
+		const tag = /^\d/.test(packageInfo.targetVersion) ? packageInfo.targetVersion : packageInfo.targetVersion.slice(1)
+		await info(`${packageInfo.name}`, `is up to date on`, `v${tag}`)
+		await sleep(random(50, 150));
+	}
+	if (toInstall.length === 0 && !ctx.dryRun) {
+		log('')
+		await success(random(celebrations), random(done));
+		return;
+	}
+	const majors: PackageInfo[] = []
+	for (const packageInfo of [...dependencies, ...devDependencies].sort(sortPackages)) {
```

Comment:
Any reason why we create **another new** array, when we already have `toInstall`? Can we reason the same array somehow?

Reply:
Nope, I think I just missed this in a refactor. Updated!